pkg/userdata/cloudinit/cloudinit.go

```diff
@@ -81,3 +103,62 @@ func buildEtcdadmArgs(config bootstrapv1alpha3.CloudInitConfig) userdata.Etcdadm
 		InstallDir:     config.InstallDir,
 	}
 }
+
+func setProxy(proxy *bootstrapv1alpha3.ProxyConfiguration, input *userdata.BaseUserData) error {
+	if proxy == nil {
+		return nil
+	}
+	tmpl := template.New("proxy").Funcs(template.FuncMap{"stringsJoin": strings.Join})
+	t, err := tmpl.Parse(proxyConf)
+	if err != nil {
+		return fmt.Errorf("failed to parse proxy template: %v", err)
+	}
+
+	var out bytes.Buffer
+	err = t.Execute(&out, proxy)
+	if err != nil {
+		return fmt.Errorf("error generating proxy config file: %v", err)
+	}
+
+	input.AdditionalFiles = append(input.AdditionalFiles, capbk.File{
+		Content: out.String(),
+		Owner:   "root:root",
+		Path:    "/etc/systemd/system/containerd.service.d/http-proxy.conf",
+	})
+
+	input.PreEtcdadmCommands = append(input.PreEtcdadmCommands, containerdRestart...)
+	return nil
+}
+
+func setRegistryMirror(registryMirror *bootstrapv1alpha3.RegistryMirrorConfiguration, input *userdata.BaseUserData) error {
+	if registryMirror == nil {
+		return nil
+	}
+	tmpl := template.New("registryMirror")
+	t, err := tmpl.Parse(registryMirrorConf)
+	if err != nil {
+		return fmt.Errorf("failed to parse registryMirror template: %v", err)
+	}
+
+	var out bytes.Buffer
+	err = t.Execute(&out, registryMirror)
+	if err != nil {
+		return fmt.Errorf("error generating registryMirror config file: %v", err)
+	}
+
+	input.AdditionalFiles = append(input.AdditionalFiles, []capbk.File{
+		{
+			Content: registryMirror.CACert,
+			Owner:   "root:root",
+			Path:    fmt.Sprintf("/etc/containerd/certs.d/%s/ca.crt", registryMirror.Endpoint),
+		}, {
+			Content: out.String(),
+			Owner:   "root:root",
+			Path:    "/etc/containerd/config_append.toml",
+		},
+	}...)
```

Comment:
why do we need to create a slice to then serialize it? what about:
```suggestion
	input.AdditionalFiles = append(input.AdditionalFiles,
		capbk.File{
			Content: registryMirror.CACert,
			Owner:   "root:root",
			Path:    fmt.Sprintf("/etc/containerd/certs.d/%s/ca.crt", registryMirror.Endpoint),
		},
		capbk.File{
			Content: out.String(),
			Owner:   "root:root",
			Path:    "/etc/containerd/config_append.toml",
		},
	)
```

Reply:
I changed it but aren't they both effectively the same?